### PR TITLE
Refactor network modification endpoints

### PR DESCRIPTION
**Pull request by flomillot**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java [hunk 1]
```diff
@@ -19,74 +19,78 @@
...
+            default:
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
         }
-        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+    }
+
+    @ExceptionHandler(ServerWebInputException.class)
+    protected ResponseEntity<Object> handleStudyException(ServerWebInputException exception) {
+        Throwable cause = exception.getCause();
+        if (cause instanceof TypeMismatchException && cause.getCause() != null && cause.getCause() != cause) {
```
Comment (archendel) [inline src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java:86]: should i be in the next handler ?

[the remark this answers is not in the document]
Comment (flomillot) [reply src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java:86]: Its the cause here, not the original exception. Did I miss something ?

Comment (flomillot) [reply src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java:86]: I don't understand ?

File: src/main/java/org/gridsuite/study/server/dto/modification/ModificationInfos.java [hunk 2]
```diff
@@ -37,5 +35,5 @@ public class ModificationInfos {
 
     @Schema(description = "Substations ID")
     @Builder.Default
-    private Set<String> substationIds = Set.of();
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/main/java/org/gridsuite/study/server/dto/modification/ModificationInfos.java:40]: Same as for the other PR I guess, I put it back

File: pom.xml [hunk 3]
```diff
@@ -291,6 +291,11 @@
             <artifactId>equalsverifier</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-contract-stub-runner</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (SlimaneAmar) [inline pom.xml:305]: Add      
```
<dependency>
            <groupId>com.github.tomakehurst</groupId>
            <artifactId>wiremock</artifactId>
            <scope>test</scope>
        </dependency>
```

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 4]
```diff
@@ -2207,4 +2206,101 @@ public void tearDown() {
...
+
+        final InputDestination input;
+
+        public SendInput(InputDestination input) {
+            this.input = input;
+        }
+
+        @Override
+        public String getName() {
+            return "input-send";
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java]: Use static

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java (hunk 4 above)
Comment (flomillot) [reply src/test/java/org/gridsuite/study/server/NetworkModificationTest.java]: Done

File: src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java:86]: My mistake, indentation in git made me misread the original code 
All good then

Comment (archendel) [reply src/main/java/org/gridsuite/study/server/RestResponseEntityExceptionHandler.java:86]: my mistake, all good here

**Review by archendel — APPROVED** [review]
Test: OK
Code: OK

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 5]
```diff
@@ -920,85 +874,46 @@ public ResponseEntity<List<String>> getAvailableSvgComponentLibraries() {
...
-    public ResponseEntity<Void> deleteModifications(@PathVariable("studyUuid") UUID studyUuid,
-                                                          @PathVariable("nodeUuid") UUID nodeUuid,
-                                                          @RequestParam(value = "modificationsUuids") List<UUID> modificationsUuids) {
+    @DeleteMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-modifications/{modificationUuids}")
+    @Operation(summary = "Delete modifications in the study network")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The network modifications was deleted"), @ApiResponse(responseCode = "404", description = "The study/node is not found")})
+    public ResponseEntity<Void> deleteNetworkModifications(@Parameter(description = "Study UUID") @PathVariable("studyUuid") UUID studyUuid,
+                                                           @Parameter(description = "Node UUID") @PathVariable("nodeUuid") UUID nodeUuid,
+                                                           @Parameter(description = "Network modification UUIDs") @PathVariable("modificationUuids") List<UUID> modificationUuids) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:914]: I think the path variable does not work for a list

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 6]
```diff
@@ -1074,91 +1009,27 @@ public void updateShortCircuitParameters(UUID studyUuid, ShortCircuitParametersE
...
             NodeModificationInfos nodeInfos = networkModificationTreeService.getNodeModificationInfos(nodeUuid);
             UUID groupUuid = nodeInfos.getModificationGroupUuid();
             String variantId = nodeInfos.getVariantId();
             UUID reportUuid = nodeInfos.getReportUuid();
             List<EquipmentModificationInfos> equipmentModificationInfosList = networkModificationService
-                    .createEquipment(studyUuid, createEquipmentAttributes, groupUuid, modificationType, variantId, reportUuid, nodeInfos.getId().toString());
-            Set<String> substationIds = getSubstationIds(equipmentModificationInfosList);
-            notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, substationIds);
-            updateStatuses(studyUuid, nodeUuid);
+                    .createModification(studyUuid, createModificationAttributes, groupUuid, modificationType, variantId, reportUuid, nodeInfos.getId().toString());
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1021]: It's not `EquipmentModificationInfos`

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**